prime-router/metadata/fhir_mapping/hl7/segments/OBX/Observation.yml

```diff
@@ -0,0 +1,490 @@
+  expressionType: JEXL
+  vars:
+    dateTimeIn: OBX.5
+    obx2: STRING, OBX.2
+_valueDateTime:
+  expressionType: nested
+  condition: $obx2 EQUALS DT || $obx2 EQUALS DTM
+  vars:
+    obx5: OBX.5
+    obx2: STRING, OBX.2
+  expressionsMap:
+    extension:
+      condition: $obx5 NOT_NULL
+      generateList: true
+      expressionType: nested
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: hl7v2-date-time
+        valueString:
+          type: STRING
+          valueOf: OBX.5
+          expressionType: HL7Spec
+
+valueTime:
+  condition: $obx2 EQUALS TM
+  type: STRING
+  valueOf: OBX.5
+  expressionType: HL7Spec
+  vars:
+    obx2: STRING, OBX.2
+
+valueRange:
+  vars:
+    obx2: STRING, OBX.2
+    CWERangeUnits: OBX.6
+    obx53: STRING, OBX.5.3
+  expressionType: nested
+  specs: OBX.5
+  expressions:
+    - condition: $obx2 EQUALS NR
+      valueOf: datatypes/NR/Range
+      expressionType: resource
+    - condition: $obx2 EQUALS SN && $obx53 EQUALS -
+      valueOf: datatypes/SN/Range
+      expressionType: resource
+
+
+valueRatio:
+  vars:
+    obx2: OBX.2
+    obx53: OBX.5.3
+    CWERangeUnits: OBX.6
+  condition: >
+    $obx2 EQUALS SN && ($obx53 EQUALS : || obx53 EQUALS /)
+  valueOf: datatypes/SN/Ratio
+  expressionType: resource
+  specs: OBX.5
+
+valueQuantity:
+  vars:
+    obx2: STRING, OBX.2
+    CWEQuantityUnits: OBX.6
+    obx53: STRING, OBX.5.3
+  expressionType: nested
+  expressions:
+    - condition: $obx2 EQUALS NM
+      expressionType: nested
+      expressionsMap:
+        value:
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: OBX.5
+        code:
+          condition: $identifier NOT_NULL && $nameOfCodingSystem NOT_NULL
+          type: STRING
+          expressionType: HL7Spec
+          valueOf: $CWEQuantityUnits.1
+          vars:
+            identifier: $CWEQuantityUnits.1
+            nameOfCodingSystem: $CWEQuantityUnits.3
+        unit:
+          expressionType: nested
+          vars:
+            identifier: $CWEQuantityUnits.1
+            text: $CWEQuantityUnits.2
+          expressions:
+            - condition: $text NULL && $identifier NOT_NULL
+              type: STRING
+              valueOf: $CWEQuantityUnits.1
+              expressionType: HL7Spec
+            - condition: $text NOT_NULL
+              type: STRING
+              valueOf: $CWEQuantityUnits.2
+              expressionType: HL7Spec
+        system:
+          condition: $identifier NOT_NULL && $nameOfCodingSystem NOT_NULL
+          type: SYSTEM_URL
+          expressionType: HL7Spec
+          valueOf: $CWEQuantityUnits.1
+          vars:
+            identifier: $CWEQuantityUnits.1
+            nameOfCodingSystem: $CWEQuantityUnits.3
+    - condition: >
+        $obx2 EQUALS SN && $obx53 NOT_EQUALS : && $obx53 NOT_EQUALS / && $obx53 NOT_EQUALS - && $obx53 NOT_EQUALS +
+      expressionType: resource
+      valueOf: datatypes/SN/Quantity
+
+referenceRange:
```

Comment:
Do you need a null check condition for `referenceRange` based on OBX.7? I think `expressionType: HL7Spec` would cover the `text` element but not the parent

Reply:
I think your right, I'll add the check